### PR TITLE
[chart] Passthrough environment variables to integrations

### PR DESCRIPTION
**Pull request by kang-makes**
With this change, users should be able to add their own custom environment variables and files so they can use them in their custom configurations.

### CODE REVIEW

File: charts/newrelic-infrastructure/templates/kubelet/_agent-config_helper.tpl [hunk 1]
```diff
@@ -6,6 +6,10 @@ http_server_enabled: true
 http_server_port: 8003
 features:
   docker_enabled: false
+{{- with .Values.kubelet.passthroughEnvironment }}
+passthrough_environment:
+  {{- toYaml . | nindent 2 }}
+{{- end }}
```
Comment (roobre) [inline charts/newrelic-infrastructure/templates/kubelet/_agent-config_helper.tpl:12]: What is the downside of specifying `passthrough_environment` in `agentConfig` directly?

Comment (roobre) [reply charts/newrelic-infrastructure/templates/kubelet/_agent-config_helper.tpl:12]: I get the side effect but, does it matter? Agents in CP/KSM scrapers run in forward mode and therefore cannot run integrations.

Comment (kang-makes) [reply charts/newrelic-infrastructure/templates/kubelet/_agent-config_helper.tpl]: I did not think about agents running in `forward_only` mode.

Anyway, I have done it in a generic way and added documentation for that.

File: charts/newrelic-infrastructure/values.yaml [hunk 2]
```diff
@@ -75,9 +75,23 @@ kubelet:
       effect: "NoExecute"
   nodeSelector: {}
   affinity: {}
+  # -- Config for the Infrastructure agent.
+  # Will be used by the forwarder sidecars and the agent running integrations.
```
Comment (roobre) [inline charts/newrelic-infrastructure/values.yaml:79]: This one will not be used by the agent running integrations, only by the control plane sidecar.

File: charts/newrelic-infrastructure/values.yaml [hunk 3]
```diff
@@ -120,6 +134,10 @@ ksm:
               matchLabels:
                 app.kubernetes.io/name: kube-state-metrics
           weight: 100
+  # -- Config for the Infrastructure agent.
+  # Will be used by the forwarder sidecars and the agent running integrations.
+  # See: https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/configuration/infrastructure-agent-configuration-settings/
+  agentConfig: {}
```
Comment (roobre) [inline charts/newrelic-infrastructure/values.yaml:140]: Ditto, only for the KSM sidecar

File: charts/newrelic-infrastructure/values.yaml [hunk 4]
```diff
@@ -188,6 +207,10 @@ controlPlane:
   # -- Run Control Plane scraper with `hostNetwork`.
   # `hostNetwork` is required for most control plane configurations, as they only accept connections from localhost.
   hostNetwork: true
+  # -- Config for the Infrastructure agent.
+  # Will be used by the forwarder sidecars and the agent running integrations.
```
Comment (roobre) [inline charts/newrelic-infrastructure/values.yaml:211]: This one is correct but I think message should be more specific so users can tell the difference between this one and the `common` one.

File: charts/newrelic-infrastructure/values.yaml (hunk 2 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/values.yaml]: Solved in [#34a64bf](https://github.com/newrelic/nri-kubernetes/pull/422/commits/34a64bfdba0341b3f159726705bcf0e54541c9f4)

File: charts/newrelic-infrastructure/values.yaml (hunk 3 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/values.yaml:139]: Solved in [#34a64bf](https://github.com/newrelic/nri-kubernetes/pull/422/commits/34a64bfdba0341b3f159726705bcf0e54541c9f4)

File: charts/newrelic-infrastructure/values.yaml (hunk 4 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/values.yaml]: I changed the wording in the comit [#34a64bf](https://github.com/newrelic/nri-kubernetes/pull/422/commits/34a64bfdba0341b3f159726705bcf0e54541c9f4)

LMKWYT

**Review by roobre — APPROVED** [review]

**Pull request merged**